docs/DeveloperGuide.adoc

```diff
@@ -364,6 +365,136 @@ We are using `java.util.logging` package for logging. The `LogsCenter` class is
+desired command in the command helper.
+
+==== Design Considerations
+
+**Aspect:** Implementation of the list view in `CommandBoxHelper` +
+**Alternative:** Instead of creating a new class to populate the ListView, use a StackPane instead of a ListView and add Strings to it +
+**Pros:** Easier to implement the display for command helper +
+**Cons:** Fragile architecture, have to change a few functions every time an extra command is added. This involves hardcoding the additional command Strings into the functions.
+Slightly harder to display selection of command in the helper box to the user.
+
+// tag::preferences[]
+=== Preferences
+
+In addition to configuration, user can customize the address book application by editing preferences. The difference between
+configurations and preferences is that configuration stores basic information requried to start the application, while
+preferences determine high level functionalities and UI styles. +
+
+==== Implementation
+The preferences are stored in `preferences.json` by default. This location is set in `config.json`. When the application
+is started, the `initPrefs` method reads the preferences file and converts it into a `UserPrefs` object using jackson-databind
+API. +
+In `UserPrefs` class, each valid preference key is defined as a field in the class, with initial value set to the default
+value of the preference. For example, the default path to the address book local storage is :
+[source,java]
+----
+private String addressBookFilePath = "data/addressbook.xml";
+----
+There are also two methods defined for each preference: `get<PreferenceKey>` and `set<PreferenceKey>`. These are used for
+reading and writing to the preference respectively. For example, for AddressBookFilePath, we have :
+[source,java]
+----
+public String getAddressBookFilePath() {
+    return addressBookFilePath;
+}
+
+public void setAddressBookFilePath(String addressBookFilePath) {
+    this.addressBookFilePath = addressBookFilePath;
+}
+----
+
+==== Command: `pref`
+The `pref` command allows users to view or edit the preferences. The details of its usage can be found in the
+<<UserGuide, UserGuide.adoc>>. The implementation of the command will be explained here. +
+When a `pref` command is entered, the `PrefCommand` will attempt to find and execute `get<PreferenceKey>`
+method in the `UserPrefs` class using Java Reflection API. If an exception is encountered, it is assumed that the method
+does not exist and the preference key entered by the user was invalid. Similarly, if two arguments were entered, `PrefCommand`
+will again try to find and execute `set<PreferenceKey>` method.
+
+==== Design Considerations
+
+**Aspect:** Implementation of `PrefCommand` +
+**Alternative 1 (current choice):** Use Reflection API +
+**Pros:** Easier to add in a new preference by changing only the `UserPrefs` class +
+**Cons:** Hard to debug since it is less intuitive +
```

Comment:
I did not check this in class, but what happens if I try to set<AB> where AB is not a PrefKey. Is it handled well?